sdk/src/Core/Amazon.Runtime/BaseLogger.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+
+namespace Amazon.Runtime
+{
+    /// <summary>
+    /// Abstract logger class, base for any custom/specific loggers.
+    /// </summary>
+    public abstract class BaseLogger
```

Comment:
While I understand wanting to expose the `Logger` class, switching it from `Internal` to `Public` means we won't be able to make any backwards incompatible changes since it would now be publicly facing. This is probably why we kept it internal in the first place. I think this is more of a design question for the team which I will bring up.